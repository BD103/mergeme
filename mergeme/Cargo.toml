[package]
name = "mergeme"
version = "0.1.0"
edition = "2024"

[features]
default = ["derive"]
derive = ["dep:mergeme_derive"]

[dependencies]
mergeme_derive = { path = "../mergeme_derive", version = "0.1.0", optional = true }
