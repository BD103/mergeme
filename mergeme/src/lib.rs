#![no_std]

#[cfg(feature = "derive")]
pub use mergeme_derive::Merge;

pub trait Merge<Partial>: Sized {
    fn merge_in_place(&mut self, other: Partial);

    fn merge(mut self, other: Partial) -> Self {
        self.merge_in_place(other);
        self
    }
}
