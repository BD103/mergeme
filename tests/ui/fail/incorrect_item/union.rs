use mergeme::Merge;

#[derive(Merge)]
#[partial(PartialChoice)]
union Choice {
    a: u32,
    b: f32,
}

fn main() {}
