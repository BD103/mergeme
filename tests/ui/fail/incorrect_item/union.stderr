error: `#[derive(Merge)]` only works on `struct`s, not `union`s
 --> tests/ui/fail/incorrect_item/union.rs:5:1
  |
5 | union Choice {
  | ^^^^^
